### PR TITLE
Update dependencies & Code Cleanup

### PR DESCRIPTION
**Pull request by Heatmanofurioso**
This PR is intended to do 2 things.
Update non-Angular dependencies in a safe way, and to do minor code cleanups.

Unit tests were run, both in head and headless mode, and the project was built and tested locally with a personal project with my own unit tests.

The bumped dependencies are the following:
CoreJs from 2.5.4 to 3.21.0
RXJS from 6.5.5 to 7.4.0 "This is the recommended version for Angular 13"
The bumped dev dependencies are the following:
Karma & Jasmine dependencies on minor bugfixes
TSNode and Types Node

None of these updates have any breaking changes from the project's usage according to what I saw on the various dependencies changelogs.

The testing function `TestBed.get` has been deprecated and switched off with `TestBed.inject`

TSLint was run in the project with our current settings, and it cleaned up some of the files "removing unnecessary commas and similar actions, fixing identations, fixing carets, etc"

We also had a stacktrace variable in the testing reports that had an invalid TS value, which I switched with a valid value according to Jasmine.
Having `displayStrackTrace: true` was invalid

The PR touches many files, but in the vast majority of it, it's cleaning code syntax or replacing the deprecated `TestBed.get` function. I feel like this PR is useful to keep the project up-to-date and to avoid it falling into deprecation with the current versions of various libs and the current good-practices of Angular

### CODE REVIEW

File: projects/ngx-google-analytics/package.json [hunk 1]
```diff
@@ -1,12 +1,12 @@
 {
   "name": "ngx-google-analytics",
-  "version": "13.0.0",
+  "version": "13.0.1",
   "description": "A simple ng-9 wrapper to load Google Analytics dependency by angular way",
   "keywords": [ "google", "google analytics", "angular", "angular-9+", "typescript" ],
   "homepage": "https://github.com/maxandriani/ngx-google-analytics",
   "license": "MIT",
   "dependencies": {
-    "tslib": "^2.0.0"
```
Comment (hakimio) [inline projects/ngx-google-analytics/package.json:9]: `tslib` should be kept at `^2.0.0`. This version range includes `2.3.1`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -21,6 +22,15 @@ I open my heart to share this component w/ you guys, buy I don't have much free
...
 I will upgrade this pack to any angular major version as soon as possible. Unfortunately I can't replicate new features to old compatibility versions. But you can fork this repo and port does features.
 
+## 13.0.1
+* Bump Karma
+* Bump Jasmine
+* Bump RXJS to 7.4.0
+* Bump Types dependencies
+* Clean up deprecated testing variables
+* Fix typos and duplicated variables
+* Run TSLint and clean up code
```
Comment (hakimio) [inline CHANGELOG.md:32]: Most of those look like `chores` and there is no need to include them in changelog.

**Comment by hakimio** [comment]
While you are at it, you could also [migrate the project from no longer maintained TSLint to ESLint](https://github.com/angular-eslint/angular-eslint#migrating-an-angular-cli-project-from-codelyzer-and-tslint).

**Comment by hakimio** [comment]
One more small thing: since the project is no longer compatible with Angular versions lower than v12, it might make sense to add Angular to peer dependencies:
```json
  "peerDependencies": {
    "@angular/common": ">=12.0.0",
    "@angular/core": ">=12.0.0"
  },
```

**Comment by Heatmanofurioso** [comment]
I didn't remember the peer dependencies. Nice notice!

Let me just test this locally on a project of mine, but I'm almost sure none of the latest commits would break something. Will ping once I've give it some tests

File: projects/ngx-google-analytics/package.json [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
+  "version": "13.0.1",
   "description": "A simple ng-9 wrapper to load Google Analytics dependency by angular way",
   "keywords": [ "google", "google analytics", "angular", "angular-9+", "typescript" ],
   "homepage": "https://github.com/maxandriani/ngx-google-analytics",
   "license": "MIT",
   "dependencies": {
     "tslib": "^2.0.0"
   },
+  "peerDependencies": {
+    "@angular/common": "^13.2.2",
```
Comment (hakimio) [inline projects/ngx-google-analytics/package.json:12]: That's a very restrictive version range. The library should work just fine with Angular v12 and higher.

Comment (Heatmanofurioso) [reply projects/ngx-google-analytics/package.json]: Updated. Thanks for the patience reviewing this
Also finished giving it a test on my side, and everything seems to be working fine

File: projects/ngx-google-analytics/package.json [hunk 4]
```diff
@@ -1,13 +1,17 @@
...
+  "version": "13.0.1",
   "description": "A simple ng-9 wrapper to load Google Analytics dependency by angular way",
   "keywords": [ "google", "google analytics", "angular", "angular-9+", "typescript" ],
   "homepage": "https://github.com/maxandriani/ngx-google-analytics",
   "license": "MIT",
   "dependencies": {
     "tslib": "^2.0.0"
   },
+  "peerDependencies": {
+    "@angular/common": "^12.0.0",
```
Comment (hakimio) [inline projects/ngx-google-analytics/package.json:12]: This version range doesn't include v13. I would recommend using the following:
```json
  "peerDependencies": {
    "@angular/common": ">=12.0.0",
    "@angular/core": ">=12.0.0"
  }
```
or if you want to be more restrictive/precise, you could go with the following:
```json
  "peerDependencies": {
    "@angular/common": ">=12.0.0 <14.0.0",
    "@angular/core": ">=12.0.0 <14.0.0"
  }
```
See [semver calculator](https://semver.npmjs.com/) for for more info.

Comment (Heatmanofurioso) [reply projects/ngx-google-analytics/package.json]: Fixed it. I decided to be a bit more precise on this, like your suggestion

**Review by maxandriani — APPROVED** [review]

**Pull request merged**